relay-server/src/actors/project.rs

```diff
@@ -506,21 +498,34 @@ impl Project {
         }
     }
 
+    pub fn merge_rate_limits(&mut self, rate_limits: RateLimits) {
+        self.rate_limits.merge(rate_limits);
+    }
+
     /// Returns a reference to the project state if available.
     pub fn state(&self) -> Option<&ProjectState> {
         self.state.as_deref()
     }
 
+    /// Returns a reference to the project state if available.
+    pub fn state_clone(&self) -> Option<Arc<ProjectState>> {
+        self.state.clone()
+    }
+
     /// Creates the aggregator if it is uninitialized and returns it.
     ///
     /// Returns `None` if the aggregator is permanently disabled, primarily for disabled projects.
-    fn get_or_create_aggregator(
+    pub fn get_or_create_aggregator(
         &mut self,
-        context: &mut Context<Self>,
+        context: &mut Context<ProjectCache>,
```

Comment:
Instead of passing context in here, would it be cleaner to pass the recipient to the constructor of `Project`?

Reply:
ok, keeping a Addr<ProjectCache> in project and using this instead of Context